cmd/cli/plugin/airgapped-network/util.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/cmd"
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/registry"
+	v1 "github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/v1"
+)
+
+var totalImgCopiedCounter int
+
+func printErrorAndExit(err error) {
+	fmt.Println(err.Error())
+	os.Exit(1)
+}
+
+func IsTagValid(tag string) bool {
+	if tag == "" {
+		return false
+	}
+
+	if !strings.HasPrefix(tag, "v") {
+		return false
+	}
+
+	tag = strings.TrimPrefix(tag, "v")
+	versions := strings.Split(tag, ".")
+	if len(versions) != 3 {
+		return false
+	}
+	for _, version := range versions {
+		if _, err := strconv.Atoi(version); err != nil {
+			return false
+		}
+	}
+	return true
+
+}
+
+func imgpkgCopyImagefromtar(sourceImageName string, destImageRepo string, customImageRepoCertificate string) {
+	confUI := ui.NewConfUI(ui.NewNoopLogger())
+	copyOptions := cmd.NewCopyOptions(confUI)
+	copyOptions.Concurrency = 1
+	copyOptions.TarFlags.TarSrc = sourceImageName
+	copyOptions.RepoDst = destImageRepo
+	if customImageRepoCertificate != "" {
+		copyOptions.RegistryFlags.CACertPaths = []string{customImageRepoCertificate}
+	}
+	err := copyOptions.Run()
+	if err != nil {
+		printErrorAndExit(errors.Wrapf(err, "failed!\n"))
+	}
+}
+
+func imgpkgCopyImagetotar(sourceImageName string, destImageRepo string) {
+	confUI := ui.NewConfUI(ui.NewNoopLogger())
+	copyOptions := cmd.NewCopyOptions(confUI)
+	copyOptions.Concurrency = 5
+	copyOptions.ImageFlags = cmd.ImageFlags{Image: sourceImageName}
+	copyOptions.TarFlags.TarDst = destImageRepo
+	err := copyOptions.Run()
+	if err != nil {
+		printErrorAndExit(errors.Wrapf(err, "failed!\n"))
+	}
+	totalImgCopiedCounter++
+}
+
+func imgpkgCopyBundletotar(sourceImageName string, destImageRepo string) {
+	confUI := ui.NewConfUI(ui.NewNoopLogger())
+	copyOptions := cmd.NewCopyOptions(confUI)
+	copyOptions.Concurrency = 5
+	copyOptions.BundleFlags = cmd.BundleFlags{Bundle: sourceImageName}
+	copyOptions.TarFlags.TarDst = destImageRepo
```

Comment:
TarFlags have a field called [Resume](https://github.com/vmware-tanzu/carvel-imgpkg/blob/develop/pkg/imgpkg/cmd/tar_flags.go#L13) that we should allow users to set when calling this plugin.
This flag will prevent images from being downloaded if they already exist in the destination folder. This can provide a huge speed up to any download and we should definitely allow the users to benefit from it.


This also applies to the function imgpkgCopyImagetotar